org.eclipse.paho.android.service/src/main/java/org/eclipse/paho/android/service/MqttAndroidClient.java

```diff
@@ -1526,6 +1528,28 @@ private void simpleAction(IMqttToken token, Bundle data) {
 		}
 	}
 
+	private void subscribeResultAction(IMqttToken token, Bundle data) {
+		if (token != null) {
+			int[] grantedQos = data.getIntArray(MqttServiceConstants.GRANTED_QOS);
+			String[] topics = token.getTopics();
+			List<String> successTopics = new ArrayList<>();
+			List<String> failedTopics = new ArrayList<>();
+			if (grantedQos != null) {
+				for (int i = 0; i < topics.length; i++) {
+					if (grantedQos[i] == 128) {

```

Comment:
Do we know for sure that `grantedQos` will be of same length as `token.getTopics()`? // just asking

Reply:
Yes it should be

Reply:
@rubenquadros12 please make that `128` as constant variable, then it would be readable.